the-collab-lab/tcl-27-smart-shopping-list
# Create addItem form, removed List.js, form updates to database

Co-authored-by: Yolanda Haynes (yhaynes@giyo.us)

## Description


This code added a form in the AddItem component that allows users to input an item which then gets submitted to the firestore database. There are also radio buttons to let the users pick the time frame of soon, kind of soon, not soon for the next time to buy the item. We also combined two components (List & GroceryContainer) into one (GroceryContainer) after conversation with the team, we all decided that both components had the same purpose so we combined them to reduce clutter and confusions.

**Display items from database**

The GrocercyContainer component will display items which grab the information from the firestore collection and display the item name, frequency (soon, kind of soon, not soon), last purchase date, and token.

**Write items to the database**
Inside the AddItem component, is a form which lets the users input what item to add to the list, when they need it next, and last purchase date which then gets created in the firestore grocery collection.

## Related Issue



Closes [#4](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/4)

## Acceptance Criteria


- User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
- When the user submits the form, the item is saved to the database, associated with the user’s token
- Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
- The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
- Item names should be displayed as a simple, unordered list on the list view after they’re added


## Type of Changes



| 	| Type                   	|
| --- | -------------------------- |
|	| :bug: Bug fix          	|
|  ✓ | :sparkles: New feature 	|
|	| :hammer: Refactoring   	|
|	| :100: Add tests        	|
|	| :link: Update dependencies |
|	| :scroll: Docs          	|

## Updates
![AddListItem](https://user-images.githubusercontent.com/58542682/126412630-01fcdf45-286f-432d-ae18-4846593f6416.gif)


## Testing Steps / QA Criteria




-  From your terminal, pull down this branch with git pull origin yh-dp-addItem-form and check that branch out with git checkout yh-dp-addItem-form
- Then yarn install or npm install the newly added dependencies locally and yarn start or npm start to launch the app.
- Fill out the form with a new item. Make sure it adds it to the list and make sure that item appears in the items collection of the database [here.](https://console.firebase.google.com/u/0/project/tcl-27-shopping-list/firestore/data/~2F)
